wgpu-hal/src/metal/adapter.rs

```diff
@@ -612,6 +612,7 @@ impl super::PrivateCapabilities {
             function_specialization: Self::supports_any(device, FUNCTION_SPECIALIZATION_SUPPORT),
             depth_clip_mode: Self::supports_any(device, DEPTH_CLIP_MODE),
             texture_cube_array: Self::supports_any(device, TEXTURE_CUBE_ARRAY_SUPPORT),
+            supports_float_filtering: device.supports_32bit_float_filtering(),
```

Comment:
```suggestion
            supports_float_filtering: os_is_mac
                || (version.at_least((11, 0), (14, 0), os_is_mac)
                    && device.supports_32bit_float_filtering()),
```

This is always available on macOS.
We also need to gate the usage of `supports32BitFloatFiltering` as it's not always available.